tests/functional-test-microceph.sh

```diff
@@ -0,0 +1,242 @@
+    snap refresh --hold microceph
+else
+    echo "MicroCeph already installed"
+fi
+
+echo "=== Step 2: Bootstrap MicroCeph cluster ==="
+if ! microceph cluster list 2>/dev/null | grep -q "$(hostname)"; then
+    echo "Bootstrapping MicroCeph cluster..."
+    microceph cluster bootstrap
+else
+    echo "MicroCeph cluster already bootstrapped"
+fi
+
+echo "=== Step 3: Add OSDs ==="
+# Check if we already have OSDs
+OSD_COUNT=$(ceph osd stat | grep -oP '\d+(?= osds:)' || echo "0")
+if [ "$OSD_COUNT" -lt 3 ]; then
+    echo "Adding 3 loop-backed OSDs (4GB each)..."
+    microceph disk add loop,4G,3
+else
+    echo "Already have $OSD_COUNT OSDs"
+fi
+
+echo "=== Step 4: Wait for cluster to be healthy ==="
+TIMEOUT=120
+ELAPSED=0
+while [ $ELAPSED -lt $TIMEOUT ]; do
+    if ceph status | grep -q "HEALTH_OK\|HEALTH_WARN"; then
+        echo "Cluster is ready"
+        break
+    fi
+    echo "Waiting for cluster to be ready... ($ELAPSED/$TIMEOUT seconds)"
+    sleep 5
+    ELAPSED=$((ELAPSED + 5))
+done
+
+ceph status
+
+echo "=== Step 5: Get Ceph version from snap manifest ==="
+CEPH_VERSION=$(cat /snap/microceph/current/snap/manifest.yaml | grep "^ceph-osd:" | awk '{print $2}')
+LIBRADOS_VERSION=$(cat /snap/microceph/current/snap/manifest.yaml | grep "^librados2:" | awk '{print $2}')
+
+echo "Ceph OSD version: $CEPH_VERSION"
+echo "Librados version: $LIBRADOS_VERSION"
+
+echo "=== Step 6: Locate DWARF JSON files in repository ==="
+# Look for matching DWARF files in the repository
+OSD_DWARF="$PROJECT_ROOT/files/ubuntu/osdtrace/osd-${CEPH_VERSION}_dwarf.json"
+RADOS_DWARF="$PROJECT_ROOT/files/ubuntu/radostrace/${LIBRADOS_VERSION}_dwarf.json"
+
+if [ ! -f "$OSD_DWARF" ]; then
+    echo "Warning: OSD DWARF file not found at $OSD_DWARF"
+    echo "Looking for any available OSD DWARF files..."
+    OSD_DWARF=$(find "$PROJECT_ROOT/files/ubuntu/osdtrace/" -name "*_dwarf.json" | head -1)
+    if [ -z "$OSD_DWARF" ]; then
+        echo "Error: No OSD DWARF files found in repository"
+        exit 1
+    fi
+    echo "Using: $OSD_DWARF"
+fi
+
+if [ ! -f "$RADOS_DWARF" ]; then
+    echo "Warning: Rados DWARF file not found at $RADOS_DWARF"
+    echo "Looking for any available radostrace DWARF files..."
+    RADOS_DWARF=$(find "$PROJECT_ROOT/files/ubuntu/radostrace/" -name "*_dwarf.json" | head -1)
+    if [ -z "$RADOS_DWARF" ]; then
+        echo "Error: No radostrace DWARF files found in repository"
+        exit 1
+    fi
+    echo "Using: $RADOS_DWARF"
+fi
+
+echo "Using OSD DWARF file: $OSD_DWARF"
+echo "Using Rados DWARF file: $RADOS_DWARF"
+
+echo "=== Step 7: Find OSD process PID ==="
+OSD_PID=$(pgrep -f "ceph-osd.*--id 0" | head -1)
+if [ -z "$OSD_PID" ]; then
+    echo "Error: Could not find ceph-osd process"
+    ps aux | grep ceph-osd
+    exit 1
+fi
+echo "Found OSD process: PID $OSD_PID"
+
+echo "=== Step 8: Create RBD pool and image for testing ==="
+# Create RBD pool if it doesn't exist
+if ! ceph osd pool ls | grep -q "^test_pool$"; then
```

Comment:
s/ceph/microceph.ceph 

and below too.

Reply:
Thanks Pon, let me try to uninstall the pre-installed ceph package and see if it works